corehq/apps/userreports/reports/specs.py

```diff
@@ -352,6 +354,45 @@ def get_query_column_ids(self):
         return [self._year_column_alias(), self._month_column_alias()]
 
 
+class ConditionalAggregationColumn(ReportColumn):
+    """
+    Used for grouping by SQL conditionals
+
+    Wraps a SQLAlchemy "case" expression:
+    http://docs.sqlalchemy.org/en/latest/core/sqlelement.html#sqlalchemy.sql.expression.case
+    """
+    type = TypeProperty('conditional_aggregation')
+    whens = DictProperty()
+    else_ = StringProperty()
+    sortable = BooleanProperty(default=False)
+
+    def get_column_config(self, data_source_config, lang):
+        return ColumnConfig(columns=[
+            DatabaseColumn(
+                header=self.get_header(lang),
+                agg_column=ConditionalAggregation(
+                    whens=self.get_whens(),
+                    else_=self.else_,
+                    alias=self.column_id,
+                ),
+                sortable=self.sortable,
+                data_slug=self.column_id,
+                format_fn=self.get_format_fn(),
+                help_text=self.description,
+                visible=self.visible,
+            )],
+        )
+
+    def get_whens(self):
+        transform_val = lambda val: "'{}'".format(val)
```

Comment:
should this be called `quote_val`? also, what happens when there's a `'` in val? is there not a util in sqlalchemy that can safely do this?